code/segmentedtickerplant/stplog.q

```diff
@@ -0,0 +1,51 @@
+// Utilites for periodic tp logging in stp process
+
+\d .stplg
+
+checkends:{
+  // jump out early if don't have to do either
+  if[nextendUTC > x; :()];
+  // when rollovermode=`period we always want to rolllogs as EOD doesn't run, only EOP
+  rolllogs:$[.finspace.rollovermode~`period;1b;not .eodtime.nextroll < x];
+  // check for endofperiod
+  if[nextperiod < x1:x+.eodtime.dailyadj;
+    stpeoperiod[.stplg`currperiod;.stplg`nextperiod;.stplg.endofdaydata[],(enlist `p)!enlist x1;rolllogs]];
+  // check for endofday
+  // In Finspace there is no need for EOD in rollovermode=`period, only EOP as there is nothing extra EOD can do that EOP doesn't
+  if[(.eodtime.nextroll < x) & not .finspace.rollovermode~`period;
+    if[.eodtime.d<("d"$x)-1;system"t 0";'"more than one day?"]; endofday[.eodtime.d;.stplg.endofdaydata[],(enlist `p)!enlist x]];
+ };
+
+init:{[dbname]
+  t::tables[`.]except `currlog;
+  msgcount::rowcount::t!count[t]#0;
+  tmpmsgcount::tmprowcount::(`symbol$())!`long$();
+  logtabs::$[multilog~`custom;key custommode;t];
+  rolltabs::$[multilog~`custom;logtabs except where custommode in `tabular`singular;t];
+  currperiod::multilogperiod xbar .z.p+.eodtime.dailyadj;
+  nextperiod::multilogperiod+currperiod;
+  getnextendUTC[];
+  i::1;
+  seqnum::0;
+  dldir::`$kdbtplog;
```

Comment:
as far as I can see, the main differences between this and the original function are this line setting `dldir` and the removal of the call to `createld` - it might be cleaner to instead redefine `createld` something like:

```
createld:{[name;date].stplg.dldir:`$.stplg.kdbtplog}
```

or if that `.stplg.kdbtplog` variable isn't used elsewhere, just remove it from config file and do

```
createld:{[name;date].stplg.dldir:`$.aws.tp_log_path}
```

Either way, this reduces the amount of code being duplicated here from TorQ repo, so less code to have to update if there are changes in TorQ.

(The other bit that has been removed from this init function is related to chained stp, not sure if there's a reason it was removed?)